images/win/scripts/Installers/Install-JavaTools.ps1

```diff
@@ -87,19 +96,37 @@ function Install-JavaFromAdoptOpenJDK {
     New-Item -ItemType File -Path $javaVersionPath -Name "$Architecture.complete" | Out-Null
 }
 
-$jdkVersions = (Get-ToolsetContent).java.versions
+$jdkVendors = (Get-ToolsetContent).java.vendors.name
+$defaultVendor = (Get-ToolsetContent).java.default_vendor
 $defaultVersion = (Get-ToolsetContent).java.default
 
-foreach ($jdkVersion in $jdkVersions) {
-    Install-JavaFromAdoptOpenJDK -JDKVersion $jdkVersion
+# We get the difference between Adopt and Adoptium and then assign it as Adopt's JAVA_HOME
+$uniqVersion = (Get-ToolsetContent).java.vendors.versions | Group-Object -noelement | Where-Object {$_.count -eq 1} | Select-Object -expand Name
+
+foreach ($jdkVendor in $jdkVendors) {
+
+    $jdkVersionsToInstall = ((Get-ToolsetContent).java.vendors | Where-Object { $_.name -eq $jdkVendor }).versions
+    $isDefaultVendor = $jdkVendor -eq $defaultVendor
+
+    foreach ($jdkVersionToInstall in $jdkVersionsToInstall)
+    {
+        Install-JavaJDK -FullVendorName $jdkVendor -JDKVersion $jdkVersionToInstall
+
+        if ($isDefaultVendor) {
```

Comment:
I think we need one more condition here.
We should invoke `Set-JavaPath` with `-DefaultVersion` only when isDefaultVendor and also isDefaultVersion.
If it is not default version, we should invoke `Set-JavaPath` without `-DefaultVersion`

Reply:
Good catch, otherwise it might indeed fail (as I see per ci)